core/src/cluster_info.rs

```diff
@@ -704,24 +730,23 @@ impl ClusterInfo {
         stakes: Option<&HashMap<Pubkey, u64>>,
     ) -> Result<()> {
         let mut last_err = Ok(());
-        let mut broadcast_table_len = 0;
+        let (peers, peers_and_stakes) = self.tvu_peers_and_stakes(stakes);
+        let broadcast_len = peers_and_stakes.len();
+        if broadcast_len == 0 {
+            datapoint_info!("cluster_info-num_nodes", ("count", 1, i64));
+            return Ok(());
+        }
         shreds.iter().zip(seeds).for_each(|(shred, seed)| {
-            let broadcast_table = self.sorted_tvu_peers(stakes, ChaChaRng::from_seed(*seed));
-            broadcast_table_len = cmp::max(broadcast_table_len, broadcast_table.len());
+            let broadcast_index = weighted_best(&peers_and_stakes, ChaChaRng::from_seed(*seed));
```

Comment:
Ah, I see we don't need to shuffle the entire list when we're only going to send it to the "best" one from the list. nice!